embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpansService.kt

```diff
@@ -34,14 +34,16 @@ internal class EmbraceSpansService(
         if (!initialized.get()) {
             synchronized(initialized) {
                 if (!initialized.get()) {
-                    currentSessionSpan.startInitialSession(sdkInitStartTimeNanos)
                     currentDelegate = SpansServiceImpl(
                         spansSink = spansSink,
                         currentSessionSpan = currentSessionSpan,
                         tracer = tracer,
                     )
-                    initialized.set(true)
-                    uninitializedSdkSpansService.recordBufferedCalls(this)
+                    currentDelegate.initializeService(sdkInitStartTimeNanos)
+                    if (currentDelegate.initialized()) {
+                        initialized.set(true)
```

Comment:
I'm not sure it makes sense to store state in this service anymore when the `SpansService` interface now contains that state. IMO we should be checking the value of `currentDelegate` and can remove the `initialized` property.